protocol/messenger_local_backup.go

```diff
@@ -0,0 +1,196 @@
+package protocol
+
+import (
+	"context"
+	crand "crypto/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+
+	gocommon "github.com/status-im/status-go/common"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+	v1protocol "github.com/status-im/status-go/protocol/v1"
+)
+
+// localBackupInterval is the duration we should allow between backups
+var localBackupInterval = 1800 * time.Second // 30 minutes
+
+func (m *Messenger) startLocalBackupLoop() {
+	ticker := time.NewTicker(localBackupInterval)
+	go func() {
+		defer gocommon.LogOnPanic()
+		for {
+			select {
+			case <-ticker.C:
+				enabled, err := m.backupEnabled()
+				if err != nil {
+					m.logger.Error("failed to fetch backup enabled")
+					continue
+				}
+				if !enabled || !m.config.featureFlags.EnableLocalBackup {
+					m.logger.Debug("backup not enabled, skipping")
+					continue
+				}
+
+				m.logger.Debug("backing up data")
+
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
otherwise the procedure will block the shutdown
```suggestion
				ctx, cancel := context.WithTimeout(m.ctx, 5*time.Minute)
```